Extract ORMs from doorkeeper, take 2

tute opened the pull request:
Related PR: https://github.com/doorkeeper-gem/doorkeeper-orms/pull/2

Idea:

* Clone down and checkout doorkeeper when `doorkeeper-orms` runs `rake`
* Configure doorkeeper to load orm's code and each of its ORMs, run
  doorkeeper tests after extending with ORMs

To do:

- [x] Move documentation removed from here into doorkeeper-orms project
- [x] `doorkeeper-orms` renamed to `doorkeeper-mongodb`
- [x] Move ORM specific generators (might be buggy)
- [x] Move ORM specific code from`spec/*` to `doorkeeper-mongodb`.
- [ ] If successful, release doorkeeper 3, document in both projects. Requires an app making sure documentation is correct.

Both projects see a good portion of their code bases removed, keeping same behavior, test coverage, and getting faster specs. WIN.

tute commented:
@jasl please tell me how you like these changes! I think this is a good way of solving this problem.

jasl commented:
Let's me check after work.

jasl commented:
@tute I comment in [doorkeeper-orms](https://github.com/doorkeeper-gem/doorkeeper-orms/pull/2/files#r29564161)

tute commented on doorkeeper.gemspec:
```diff
@@ -26,6 +26,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency "database_cleaner", "~> 1.3.0"
   s.add_development_dependency "rspec-activemodel-mocks", "~> 1.0.0"
-  s.add_development_dependency "bcrypt-ruby", "~> 3.0.1"
   s.add_development_dependency "pry", "~> 0.10.0"
```
@jasl can we remove dependencies from here?

jasl replied:
I think yes

tute commented:
It looks both PRs are green. Updated documentation in both repos, and renamed the mongodb extensions one.

We just need a real mongo doorkeeper application to make sure everything has been tied together correctly. Do you have anyone at hand, @jasl? Thank you!

jasl commented:
No problem, I will update our application to doorkeeper edge version and test

jasl commented:
We have update our developing branch using this gem, if have problem, I will fix

jasl commented:
oh I mean doorkeeper-mongo

tute commented:
I created a new `2.2-stable` branch that will receive security fixes from now on. master is already tracking version 3, which I expect to release whenever we see a mongodb doorkeeper v3 installation working correctly, and making sure documentation is correct.

Thank you!

tute commented:
In preparation for Rails 3, if we still have changes we didn't do due to backwards compatibility concerns, or we have leftover upgrading warnings, we should change/remove those now.

tute commented:
@jasl could you test doorkeeper 3 with your MongoDB project, so we make sure it works and can release v3 stable?

jasl commented:
OK, give me a week to test it stable.
